### PR TITLE
Add team preferences functionality to provider

### PR DESCRIPTION
**Pull request by jonathan-dorsey**
This pull request will add the ability to maintain team preferences in the Grafana provider. The largest driver for this change is to be able to specify a home dashboard for each team. This PR is similar in structure to the [PR](https://github.com/grafana/terraform-provider-grafana/pull/128) that added folder permissions. Since team preferences cannot be added or deleted, we always perform an update. The delete will simply attempt to use the system default values, but if the corresponding team has already been deleted we simply ignore the error and continue.

### CODE REVIEW

**Comment by jonathan-dorsey** [comment]
NOTE: This PR cannot be merged until https://github.com/grafana/grafana-api-golang-client/pull/11/files is merged for the Go API. The call to update the preferences was incorrect and needed to be fixed. Once that PR is merged I will update the version of the api being referenced and call the "go mod vendor" and push those changes.

**Comment by jonathan-dorsey** [comment]
@trotttrotttrott  - I believe this PR is ready for review now. Thanks!

File: website/docs/r/team_preferences.html.md [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+The team preferences resource allows for team preferences to be set once a team 
+has been created. Available preferences are a light or dark theme, the default
+timezone to be used, and the dashboard to be displayed upon login. 
+
+## Example Usage
+
+```hcl
+resource "grafana_dashboard" "metrics" {
+  config_json = "${file("grafana-dashboard.json")}"
```
Comment (trotttrotttrott) [inline website/docs/r/team_preferences.html.md]:
```suggestion
  config_json = file("grafana-dashboard.json")
```

File: website/docs/r/team_preferences.html.md [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+}
+
+resource "grafana_team" "team" {
+  name = "Team Name"
+}
+
+resource "grafana_team_preferences" "teamPreferences" {
```
Comment (trotttrotttrott) [inline website/docs/r/team_preferences.html.md]: Would like to have all example resource names use underscores.
```suggestion
resource "grafana_team_preferences" "team_preferences" {
```

**Review by trotttrotttrott — COMMENTED** [review]
Looks great. Just some minor nitpicks in the docs. Could you also run `go mod tidy`?

**Comment by jonathan-dorsey** [comment]
@trotttrotttrott - comments incorporated. Thanks!

**Review by trotttrotttrott — APPROVED** [review]
Thanks once again @jonathan-dorsey!

**Pull request merged**